include/xhyve/support/misc.h

```diff
@@ -65,3 +68,25 @@ static inline void do_cpuid(unsigned ax, unsigned *p) {
 
 /* Used to trigger a self-shutdown */
 extern void push_power_button(void);
+
+/* Error checking pthread mutex operations */
+static inline void xpthread_mutex_init(pthread_mutex_t *mutex)
+{
+	int rc = pthread_mutex_init(mutex, NULL);
+	assert(rc == 0);
```

Comment:
I lazily looked at Linux's `pthread_mutex_init(1)` (which says init only ever returns 0) since it was closer to hand rather than going to the source http://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_mutex_init.html for the actual behaviour. I'll update to follow the same pattern as the other two wrappers.